java/src/jmri/jmrix/loconet/logixng/SetSpeedZero.java

```diff
@@ -0,0 +1,115 @@
+package jmri.jmrix.loconet.logixng;
+
+import jmri.jmrit.logixng.actions.*;
+
+import java.util.*;
+
+import jmri.*;
+import jmri.JmriException;
+import jmri.jmrit.logixng.*;
+import jmri.jmrix.loconet.*;
+
+/**
+ * Sets the speed to zero if the loco hasn't been used in a while.
+ *
+ * @author Daniel Bergqvist Copyright 2020
+ */
+public class SetSpeedZero extends AbstractDigitalAction {
+
+    private static final int NUM_LOCO_SLOTS_TO_CHECK = 119;
+
+    private LocoNetSystemConnectionMemo _memo;
+
+    public SetSpeedZero(String sys, String user, LocoNetSystemConnectionMemo memo) {
+        super(sys, user);
+        _memo = memo;
+    }
+
+    @Override
+    public Base getDeepCopy(Map<String, String> systemNames, Map<String, String> userNames) throws JmriException {
+        DigitalActionManager manager = InstanceManager.getDefault(DigitalActionManager.class);
+        String sysName = systemNames.get(getSystemName());
+        String userName = userNames.get(getSystemName());
+        if (sysName == null) sysName = manager.getAutoSystemName();
+        SetSpeedZero copy = new SetSpeedZero(sysName, userName, _memo);
+        return manager.registerAction(copy).deepCopyChildren(this, systemNames, userNames);
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public Category getCategory() {
+        return CategoryLocoNet.LOCONET;
+    }
+
+    public void setMemo(LocoNetSystemConnectionMemo memo) {
+        assertListenersAreNotRegistered(log, "setMemo");
+        _memo = memo;
+    }
+
+    public LocoNetSystemConnectionMemo getMemo() {
+        return _memo;
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public void execute() {
+        for (int i=1; i <= NUM_LOCO_SLOTS_TO_CHECK; i++) {
+            LocoNetSlot slot = _memo.getSlotManager().slot(i);
+            LocoNetMessage msg = new LocoNetMessage(4);
+            msg.setOpCode(LnConstants.OPC_LOCO_SPD);
```

Comment:
This is not brutal, its a disaster.
Never set a slot speed unless you know it can or needs to be reset.
Never reset the slot speed for a member of a consist.
Never reset a slot speed for an idle/free slot as in certain command stations it can cause the slot status the change.
You wont have to poweroff and on the command station, you will have to reset it!

Reply:
Thank you. I apologize and I have updated the code. Is there anything more I need to check?